### ISSUE TITLE
`navigateTo` with `external: true` causes hard reload

### ISSUE THREAD

**Issue by some-user123** [nuxt/nuxt#20856]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v16.14.2
- Nuxt Version:     3.4.3
- Nitro Version:    2.4.0
- Package Manager:  npm@7.17.0
- Builder:          vite
- User Config:      -
- Runtime Modules:  -
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/nuxt-starter-gmsge9?file=pages%2Findex.vue,pages%2Fpage2.vue

### Describe the bug

`index.vue` uses `navigateTo` without `external: true`:
- Links to internal URLs (other page) work well ✅
- Links to external URLs don't work (as expected) ✅

`page2.vue` uses `navigateTo` with `external: true`:
- Links to internal URLs (other page) cause a hard reload ❌
- Links to external URLs work ✅

I didn't expect that from the documentation (https://nuxt.com/docs/api/utils/navigate-to). From my perspective for internal URLs the value of `external` should not make any difference.
In other words: `navigateTo` with `external: true` should not cause a hard reload when used with an internal URL.

**Comment by danielroe** [comment]
I think we can improve the documentation then. This is the intended behaviour.

**Comment by some-user123** [comment]
That would solve the inconsistency, but not my problem 😉

Wouldn't it be nice(r) to have a function (`navigateTo`) that behaves like the `NuxtLink` that also won't case a hard reload on a internal URL?
